move Tar Dataset to Tar DataPipe

glaringlee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51398 move Tar Dataset to Tar DataPipe**

This PR is to make previously called TarDataset to Tar DataPipe and apply the new model structure.

Differential Revision: [D26162319](https://our.internmc.facebook.com/intern/diff/D26162319)

Commit:
move Tar Dataset to Tar DataPipe

[ghstack-poisoned]

ejguan reviewed (approved):
LGTM

ejguan commented on torch/utils/data/datapipes/iter/readfilesfromtar.py:
```diff
@@ -0,0 +1,60 @@
+from torch.utils.data.dataset import IterableDataset as IterDataPipe
```
I find an easy way just by aliasing `IterableDataset` as `IterDataPipe` in the `__init__.py` under data, then directly import IterDataPipe.

ejguan replied:
PR #51488

glaringlee replied:
Great! Updated

Commit:
Update on "move Tar Dataset to Tar DataPipe"


This PR is to make previously called TarDataset to Tar DataPipe and apply the new model structure.

Differential Revision: [D26162319](https://our.internmc.facebook.com/intern/diff/D26162319)

[ghstack-poisoned]
